kratos/tests/test_matrix_interface.py

```diff
@@ -120,5 +120,32 @@ def test_matrix_transpose(self):
             for j in range(A.Size2()):
                 self.assertEqual(A[i,j], B[j,i])
 
+    def test_list_of_list_construction(self):
+        A = KM.Matrix([[2, 2, 2, 2]])                           #List 1
+        self.assertEqual(4, len(A))
+        self.assertEqual(1, A.Size1())
+        self.assertEqual(4, A.Size2())
+        expected = KM.Matrix(1,4)
+        expected.fill(2)
+
+        for i in range(expected.Size1()):
+            for j in range(expected.Size2()):
+                self.assertEqual(A[i,j], 2)
+
+        l2 = [[0, 0.1, 0.2], [1, 1.1, 1.2], [2, 2.1, 2.2]]       #List 2
+        B = KM.Matrix(l2)
+        self.assertEqual(9, len(B))
+        self.assertEqual(3, B.Size1())
+        self.assertEqual(3, B.Size2())
+        expected = KM.Matrix(3,3)
+        for i in range(expected.Size1()):
+            for j in range(expected.Size2()):
+                expected[i,j] = i + (j*0.1)

```

Comment:
constructing this seems overhead and is confusing, can you do it without please?

Reply:
Yes, I will do tht..